app-ios/tutanota/Sources/Contacts/TutaContactFacade.swift

```diff
@@ -0,0 +1,143 @@
+			TUTSLog("Migrating old unstable hashes")
+			// Map old values Int64 to a truncated UInt32 hash
+			dict["localContactIdentifierToHash"] = (dict["localContactIdentifierToHash"] as! [String: Int]).mapValues { UInt32($0 & 0xFFFFFFFF) }
+			dict["stablehash"] = true
+		}
+		let mapping = try! DictionaryDecoder().decode(UserContactMapping.self, from: dict)
+		return try! UserContactList(nativeContactStoreFacade: self.nativeContactStoreFacade, username: username, mappingData: mapping, stableHash: true)
+	}
+	/// Returns a contact list from the mappings dictionary and migrates it if necessary
+	func getContactList(username: String) -> UserContactList? {
+		if var dict = getMappingsDictionary()[username] { return readContactList(&dict, username) } else { return nil }
+	}
+
+	func getContactLists() -> [UserContactList] {
+		getMappingsDictionary()
+			.map({ (key: String, value: [String: Any]) in
+				var dict = value
+				return readContactList(&dict, key)
+			})
+	}
+
+	/// Inserts a contact list into the mappings dictionary
+	func saveContactList(_ contactList: UserContactList) {
+		var dict = getMappingsDictionary()
+		let mapping = contactList.getMapping()
+		dict[mapping.username] = try! DictionaryEncoder().encode(mapping)
+		self.userDefaults.setValue(dict, forKey: CONTACTS_MAPPINGS)
+	}
+
+	/// Removes a contact list from the mappings dictionary and deletes the associated group from the contact store
+	func deleteContactList(_ contactList: UserContactList) throws {
+		let group = try contactList.getTutaContactGroup()
+		var dict = getMappingsDictionary()
+		let mapping = contactList.getMapping()
+		dict.removeValue(forKey: mapping.username)
+		self.userDefaults.setValue(dict, forKey: CONTACTS_MAPPINGS)
+		try self.nativeContactStoreFacade.deleteCNGroup(forGroup: group)
+	}
+
+	func getOrCreateContactList(username: String) throws -> UserContactList {
+		if let mapping = self.getContactList(username: username) {
+			return mapping
+		} else {
+			TUTSLog("MobileContactsFacade: creating new mapping for \(username)")
+			let mapping = try UserContactList(nativeContactStoreFacade: self.nativeContactStoreFacade, username: username, mappingData: nil, stableHash: true)
+			self.saveContactList(mapping)
+			return mapping
+		}
+	}
+
+	/// Returns all contacts in the iOS contact store
+	func getAllContacts(withSorting desiredSorting: CNContactSortOrder?) throws -> [StructuredContact] {
+		try nativeContactStoreFacade.getAllContacts(inGroup: nil, withSorting: desiredSorting)
+			.map { nativeContact in nativeContact.toStructuredContact(serverId: nil) }
+	}
+
+	func queryContactSuggestions(query: String, upTo: Int) throws -> [ContactSuggestion] {
+		try nativeContactStoreFacade.queryContactSuggestions(query: query, upTo: upTo)
+	}
+
+	/// Returns any contacts outside the Tuta contact list specified by `username` that match contacts inside that Tuta contact list
+	func getDuplicateContacts(_ username: String) async throws -> [StructuredContact] {
```

Comment:
it looks like this one is unused? And is maybe superseded by findDuplicateLocalContacts?